### PR TITLE
Update Rel8 to work with Opaleye's #586

### PR DESCRIPTION
**Pull request by shane-circuithub**

### CODE REVIEW

File: src/Rel8/Query/Aggregate.hs [hunk 1]
```diff
@@ -15,36 +16,44 @@ import Data.Int ( Int64 )
...
+aggregateU unpack (Aggregator _ aggregator) =
+  mapOpaleye (Opaleye.aggregateExplicit unpack aggregator)
 
 
 -- | Count the number of rows returned by a query. Note that this is different
 -- from @countStar@, as even if the given query yields no rows, @countRows@
 -- will return @0@.
 countRows :: Query a -> Query (Expr Int64)
-countRows = aggregate countStar
+countRows = aggregate countStar . (true <$)
```
Comment (tomjaguarpaw) [inline src/Rel8/Query/Aggregate.hs:59]: I'm curious about why this change is needed. What's going on here?

Comment (tomjaguarpaw) [reply src/Rel8/Query/Aggregate.hs:59]: It makes me concerned that there's some implication of the corresponding Opaleye change that I'm not aware of.

Comment (tomjaguarpaw) [reply src/Rel8/Query/Aggregate.hs:59]: Oh, maybe it's just to get a `Table Expr i` instance? If so then that's fine. (I was worried that there was some subtlety about the SQL that we generate, and this change was to avoid a crashing query.)

Comment (ocharles) [reply src/Rel8/Query/Aggregate.hs:59]: I imagine it's to avoid propagating that constraint

Comment (tomjaguarpaw) [reply src/Rel8/Query/Aggregate.hs:59]: Great, that's reassuring. If so then there's nothing I don't understand and my PR is ready to go Opaleye-side. @duairc, could you please confirm?

Comment (shane-circuithub) [reply src/Rel8/Query/Aggregate.hs:59]: Apologies for the delay, but yes I can confirm that this is just to get a `Table Expr` instance.
